File: golang/tests/pkg/integration/computer_beta_screenshot_integration_test.go
package integration

import (
	"bytes"
	"os"
	"testing"
	"time"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
	"github.com/stretchr/testify/require"
)

func TestComputerBetaScreenshotJPEG(t *testing.T) {
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		t.Skip("Skipping integration test: AGENTBAY_API_KEY not set")
	}

	agentBay, err := agentbay.NewAgentBay(apiKey)
	require.NoError(t, err)

	sessionResult, err := agentBay.Create(&agentbay.CreateSessionParams{ImageId: "linux_latest"})
	require.NoError(t, err)
	require.NotNil(t, sessionResult.Session)
	session := sessionResult.Session
	defer func() { _, _ = session.Delete() }()

	time.Sleep(10 * time.Second)

	s := session.Computer.BetaTakeScreenshot("jpg")
	require.True(t, s.Success, "beta screenshot failed: %s", s.ErrorMessage)
	require.Equal(t, "jpeg", s.Format)
	require.Greater(t, len(s.Data), 3)
	require.True(t, bytes.HasPrefix(s.Data, []byte{0xff, 0xd8, 0xff}))
}

File: typescript/tests/integration/computer-beta-screenshot.integration.test.ts
/**
 * Integration tests for Computer beta screenshot APIs.
 *
 * Note: This test creates a real session. Do not run concurrently.
 */

import { AgentBay } from "../../src/agent-bay";
import { log } from "../../src/utils/logger";

describe("Computer beta screenshot integration tests", () => {
  const apiKey = process.env.AGENTBAY_API_KEY;
  let agentBay: AgentBay;

  beforeEach(() => {
    if (apiKey) {
      agentBay = new AgentBay({ apiKey });
    }
  });

  test("should capture screenshot as JPEG bytes", async () => {
    if (!apiKey) {
      log("Skipping test: AGENTBAY_API_KEY not set");
      return;
    }

    const sessionResult = await agentBay.create({ imageId: "linux_latest" });
    expect(sessionResult.session).toBeDefined();
    const session = sessionResult.session!;

    try {
      await new Promise((resolve) => setTimeout(resolve, 10000));
      const s = await session.computer.betaTakeScreenshot("jpg");
      expect(s.success).toBe(true);
      expect(s.format).toBe("jpeg");
      expect(s.data.length).toBeGreaterThan(3);
      expect(Buffer.from(s.data).slice(0, 3).equals(Buffer.from([0xff, 0xd8, 0xff]))).toBe(true);
      expect(typeof s.width).toBe("number");
      expect(typeof s.height).toBe("number");
      expect((s.width as number) > 0).toBe(true);
      expect((s.height as number) > 0).toBe(true);
    } finally {
      await session.delete();
    }
  }, 60000);
});

